### PR TITLE
CustomerID Removed. Check Login Details.

### PR DESCRIPTION
**Pull request by sg2550**
CustomerID was removed. Username and Password instaces were added in the Customer class. Login verification method was added in the NewBank class (checKLogInDetails). Initial interface was created.

### CODE REVIEW

File: server/Account.java [hunk 1]
```diff
@@ -19,12 +19,12 @@ public String getAccountName() {
 		return accountName;
 	}
 
-	//add getter for opening balance
+	// Adding getter for opening balance.
```
Comment (MattKomorcec) [inline server/Account.java]: I'd say the comments in this file are not needed, as the methods are descriptive enough, but it's a nitpick

File: server/AccountManagement.java [hunk 2]
```diff
@@ -29,14 +29,15 @@ public String newAccount(CustomerID customer, NewBankClientHandler newBankClient
...
                 valid = true;
-            }else{
+            }
+            else{
                 newBankClientHandler.sendOutput("Account name is taken, please try again.");
             }
         }
         //creates new account, defaults to balance of 0.0
-        NewBank.getBank().getCustomers().get(customer.getKey()).addAccount(new Account(newAccountName, 0.0));
+        customer.addAccount(new newbank.server.Account(newAccountName, 0.0));
```
Comment (MattKomorcec) [inline server/AccountManagement.java]: `customer.addAccount(new newbank.server.Account(newAccountName, 0.0));` can be changed to just `customer.addAccount(new Account(newAccountName, 0.0));`

The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it everywhere in this pull request

File: server/Customer.java [hunk 3]
```diff
@@ -4,15 +4,19 @@
 
 public class Customer {
 
-	private ArrayList<Account> accounts;
+	private ArrayList<newbank.server.Account> accounts;
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Customer.java [hunk 4]
```diff
@@ -4,15 +4,19 @@
...
+	public Customer(String username, String password) {
 		accounts = new ArrayList<>();
+		this.username = username;
+		this.password = password;
 	}
 
 	public String accountsToString() {
 		String s = "";
-		for (Account a : accounts) {
+		for (newbank.server.Account a : accounts) {
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Customer.java [hunk 5]
```diff
@@ -22,30 +26,37 @@ public String accountsToString() {
 		return s;
 	}
 
-	public void addAccount(Account account) {
+	public void addAccount(newbank.server.Account account) {
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Customer.java [hunk 6]
```diff
@@ -22,30 +26,37 @@ public String accountsToString() {
...
 		accounts.add(account);
 	}
 
-	public ArrayList<Account> getAccounts() {
+	public ArrayList<newbank.server.Account> getAccounts() {
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Customer.java [hunk 7]
```diff
@@ -22,30 +26,37 @@ public String accountsToString() {
...
 		return accounts;
 	}
 
 	public boolean checkExistingAccount(String accountName) {
-		for (Account a : accounts) {
+		for (newbank.server.Account a : accounts) {
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Customer.java [hunk 8]
```diff
@@ -22,30 +26,37 @@ public String accountsToString() {
...
 			if (a.getAccountName().equals(accountName)) {
 				return true;
 			}
 		}
 		return false;
 	}
 
-	public Account getExistingAccount(String accountName) {
-		for (Account a : accounts) {
+	public newbank.server.Account getExistingAccount(String accountName) {
```
Comment (MattKomorcec) [inline server/Customer.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 9]
```diff
@@ -6,11 +6,12 @@
 public class NewBank {
 
 	private static final NewBank bank = new NewBank();
-	private HashMap<String, Customer> customers;
+	private HashMap<String, newbank.server.Customer> customers;
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 10]
```diff
@@ -6,11 +6,12 @@
...
 
-	//initialised AccManagement Object
-	private AccountManagement accountManagement = new AccountManagement();
-	//initialised Transaction Object
+	//Initialising AccountManagement Object
+	private newbank.server.AccountManagement accountManagement = new newbank.server.AccountManagement();
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 11]
```diff
@@ -19,33 +20,45 @@ private NewBank() {
 	}
 
 	private void addTestData() {
-		Customer bhagy = new Customer();
-		bhagy.addAccount(new Account("Main", 1000.0));
-		customers.put("Bhagy", bhagy);
+		newbank.server.Customer bhagy = new newbank.server.Customer("Bhagy", "1234");
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 12]
```diff
@@ -19,33 +20,45 @@ private NewBank() {
...
 	}
 
 	public static NewBank getBank() {
 		return bank;
 	}
 
-	public synchronized CustomerID checkLogInDetails(String userName, String password) {
-		if (customers.containsKey(userName)) {
-			return new CustomerID(userName);
+	public synchronized newbank.server.Customer checkLogInDetails(String username, String password, NewBankClientHandler newBankClientHandler) {
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 13]
```diff
@@ -19,33 +20,45 @@ private NewBank() {
...
+			return null;
 		}
-		return null;
 	}
 
-	// commands from the NewBank customer are processed in this method
-	public synchronized String processRequest(CustomerID customer, String request, NewBankClientHandler newBankClientHandler) {
-		if (customers.containsKey(customer.getKey())) {
+	// Commands from the NewBank customer are processed in this method.
+	public synchronized String processRequest(newbank.server.Customer customer, String request, NewBankClientHandler newBankClientHandler) {
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 14]
```diff
@@ -62,12 +75,12 @@ public synchronized String processRequest(CustomerID customer, String request, N
 		return "FAIL";
 	}
 
-	public HashMap<String, Customer> getCustomers() {
+	public HashMap<String, newbank.server.Customer> getCustomers() {
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBank.java [hunk 15]
```diff
@@ -62,12 +75,12 @@ public synchronized String processRequest(CustomerID customer, String request, N
...
 		return customers;
 	}
 
-	public String getID(Customer c) {
-		for (Entry<String, Customer> entry : customers.entrySet()) {
+	public String getID(newbank.server.Customer c) {
```
Comment (MattKomorcec) [inline server/NewBank.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBankClientHandler.java [hunk 16]
```diff
@@ -8,56 +8,84 @@
 
 public class NewBankClientHandler extends Thread {
 
-	private NewBank bank;
+	private newbank.server.NewBank bank;
```
Comment (MattKomorcec) [inline server/NewBankClientHandler.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBankClientHandler.java [hunk 17]
```diff
@@ -8,56 +8,84 @@
...
 	private BufferedReader in;
 	private PrintWriter out;
 
 	public NewBankClientHandler(Socket s) throws IOException {
-		bank = NewBank.getBank();
+		bank = newbank.server.NewBank.getBank();
```
Comment (MattKomorcec) [inline server/NewBankClientHandler.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/NewBankClientHandler.java [hunk 18]
```diff
@@ -8,56 +8,84 @@
...
+					case "1":
+						// Ask for username
+						out.println("Enter Username:");
+						String userName = in.readLine();
+						// Ask for password
+						out.println("Enter Password:");
+						String password = in.readLine();
+						out.println("\nChecking Details...");
+						// Authenticate user and get customer ID token from bank for use in subsequent requests.
+						newbank.server.Customer customer = bank.checkLogInDetails(userName, password, this);
```
Comment (MattKomorcec) [inline server/NewBankClientHandler.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Transaction.java [hunk 19]
```diff
@@ -6,186 +6,168 @@
     String confirmation = "";
     double amount = 0.0;
     boolean valid = false;
-    Account fromAccount;
-    Account toAccount;
-    Customer toCustomer;
+    newbank.server.Account fromAccount;
```
Comment (MattKomorcec) [inline server/Transaction.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

File: server/Transaction.java [hunk 20]
```diff
@@ -6,186 +6,168 @@
...
+    newbank.server.Account toAccount;
+    newbank.server.Customer toCustomer;
 
-    //move funds between personal accounts
-    public String moveFunds(CustomerID customer, NewBankClientHandler newBankClientHandler) {
+    // Moving funds between personal accounts.
+    public String moveFunds(newbank.server.Customer customer, newbank.server.NewBankClientHandler newBankClientHandler) {
```
Comment (MattKomorcec) [inline server/Transaction.java]: The `newbank.server` is not necessary, and actually makes the code harder to read, please remove it

**Review by MattKomorcec — COMMENTED** [review]
Please fix `newbank.server` where needed :)

**Comment by sg2550** [comment]
Hey Mat.

Not sure how or why this "newbank.server" was added. I did not include it in my code. I believe I have addressed it now. Please have a look when you are free.

Thanks a lot - Spyros :)

**Review by phgage — APPROVED** [review]
Looking good, looks like all of Mat's comments were addressed

**Pull request merged**